oidc-api/src/main/java/uk/gov/di/authentication/oidc/entity/LogoutRequest.java

```diff
@@ -0,0 +1,222 @@
+            rpPairwiseId = Optional.ofNullable(idToken.getJWTClaimsSet().getSubject());
+            var clientSessionId = idToken.getJWTClaimsSet().getStringClaim("sid");
+            auditUser =
+                    Objects.nonNull(clientSessionId)
+                            ? auditUser.withGovukSigninJourneyId(clientSessionId)
+                            : auditUser;
+        } catch (ParseException e) {
+            LOG.warn("Unable to extract JWTClaimsSet to get the audience");
+            errorObject =
+                    Optional.of(
+                            new ErrorObject(
+                                    OAuth2Error.INVALID_REQUEST_CODE, "invalid id_token_hint"));
+            return;
+        }
+
+        if (clientId.isEmpty() || rpPairwiseId.isEmpty()) {
+            return;
+        }
+
+        LOG.info("Validating client ID");
+        attachLogFieldToLogs(CLIENT_ID, clientId.get());
+        clientRegistry = dynamoClientService.getClient(clientId.get());
+        if (clientRegistry.isEmpty()) {
+            LOG.warn("Client not found in client registry");
+            errorObject =
+                    Optional.of(
+                            new ErrorObject(
+                                    OAuth2Error.UNAUTHORIZED_CLIENT_CODE, "client not found"));
+            return;
+        }
+
+        postLogoutRedirectUri =
+                Optional.ofNullable(queryStringParameters.get().get("post_logout_redirect_uri"));
+        if (postLogoutRedirectUri.isEmpty()) {
+            LOG.info("Post logout redirect URI not present in logout request");
+            return;
+        }
+
+        if (!postLogoutRedirectUriInClientReg()) {
+            errorObject =
+                    Optional.of(
+                            new ErrorObject(
+                                    OAuth2Error.INVALID_REQUEST_CODE,
+                                    "client registry does not contain post_logout_redirect_uri"));
+        }
+    }
+
+    private boolean postLogoutRedirectUriInClientReg() {
+        return postLogoutRedirectUri
+                .map(
+                        uri -> {
+                            if (!clientRegistry.get().getPostLogoutRedirectUrls().contains(uri)) {
+                                LOG.warn(
+                                        "Client registry does not contain the post logout redirect URI which was sent in the logout request. Value is {}",
+                                        uri);
+                                return false;
+                            } else {
+                                LOG.info(
+                                        "Post logout redirect URI is present in logout request and client registry. Value is {}",
+                                        uri);
+                                return true;
+                            }
+                        })
+                .orElseGet(() -> false);
```

Comment:
Is there a reason we can't just use
```suggestion
                .orElse(false);
```
?

Reply:
Yep can simplify